monkey/tests/unit_tests/infection_monkey/test_send_all_events_to_island.py

```diff
@@ -0,0 +1,47 @@
+import time
+
+import pytest
+import requests_mock
+
+from infection_monkey.send_all_events_to_island import EVENTS_API_URL, AgentEventsToIslandSender
+
+SERVER = "1.1.1.1:9999"
+
+
+@pytest.fixture
+def event_sender():
+    return AgentEventsToIslandSender(SERVER, time_period=0.001)
+
+
+def test_send_events(event_sender):
+    with requests_mock.Mocker() as mock:
+        mock.post(EVENTS_API_URL % SERVER)
+
+        event_sender.start()
+
+        for _ in range(5):
+            event_sender.add_event_to_queue({})
+        time.sleep(0.1)
+        assert mock.call_count == 1
+
+        event_sender.add_event_to_queue({})
+        time.sleep(0.1)
+        assert mock.call_count == 2
+
+        event_sender.stop()
+
+
+def test_send_remaining_events(event_sender):
+    with requests_mock.Mocker() as mock:
+        mock.post(EVENTS_API_URL % SERVER)
+
+        event_sender.start()
+
+        for _ in range(5):
+            event_sender.add_event_to_queue({})
+        time.sleep(0.1)
```

Comment:
```suggestion
        time.sleep(0.005)
```

Reply:
0.005 is too less. The test fails. Reduced it to 0.5.

Reply:
What about 0.01? That's 10x the sleep period, which should be enough.